### PR TITLE
feat(textbox): reduce padding & add `inputProps`

### PR DESCRIPTION
**Pull request by QuiiBz**
## Type

- Feature

#### What is expected?

The padding-right has been reduced to `required` TextBox, and this PR adds an `inputProps` prop to TextBox which are forwarded to the input element.

## Relevant logs and/or screenshots

Page | Before | After
:-   |  :-:   | -:
-- | Screenshot 2021-09-16 at 15 14 34 | Screenshot 2021-09-16 at 15 15 00

### CODE REVIEW

File: src/components/TextBox/index.tsx [hunk 1]
```diff
@@ -471,6 +480,14 @@ const TextBox = forwardRef<
     const hasRightElement =
       valid || required || isPassToggleable || random || unit
 
+    const getRightElementPadding = () => {
+      if (required && hasRightElement) {
+        return 22
```
Comment (chambo-e) [inline src/components/TextBox/index.tsx:485]: What is this number ? 🤔

File: src/components/TextBox/index.tsx [hunk 2]
```diff
@@ -618,6 +637,7 @@ TextBox.propTypes = {
   generated: PropTypes.bool,
   height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
   id: PropTypes.string,
+  inputProps: PropTypes.objectOf(PropTypes.any),
```
Comment (chambo-e) [inline src/components/TextBox/index.tsx:640]: In which case are we going to use this props ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: src/components/TextBox/index.tsx (hunk 1 above)
Comment (QuiiBz) [reply src/components/TextBox/index.tsx:485]: It allows to have the required icon centered:

Screenshot 2021-09-16 at 15 15 00

File: src/components/TextBox/index.tsx (hunk 2 above)
Comment (QuiiBz) [reply src/components/TextBox/index.tsx:640]: That was requested in CNS-3912, to reduce the input height.

File: src/components/TextBox/index.tsx (hunk 1 above)
Comment (chambo-e) [reply src/components/TextBox/index.tsx:485]: Possible to put it in a const describing it pls ? :)

**Review by chambo-e — APPROVED** [review]

**Review by DorianMaliszewski — APPROVED** [review]

**Pull request merged**